src/findOrCreateProfile.ts

```diff
@@ -1,20 +1,31 @@
 import { Request, Response } from 'express'
+import { Maybe } from 'graphql-tools'
 import { authenticateRequest } from './authenticateRequest'
+import Group from './models/group'
 import UserAccount from './models/user_account'
 
 const sendProfile = (
   response: Response,
   userAccount: UserAccount,
   isAdmin = false,
-) => response.json(userAccount.asProfile(isAdmin)).end()
+  groupId?: number,
+) => response.json(userAccount.asProfile(isAdmin, groupId)).end()
 
 const findOrCreateProfile = async (request: Request, response: Response) => {
   const auth = await authenticateRequest(request)
 
   if (auth.userAccount != null) {
     console.info('found account for profile', auth.claims.sub)
 
-    sendProfile(response, auth.userAccount, auth.isAdmin)
+    let groupForUser: Maybe<Group>
+    if (!auth.isAdmin) {
+      groupForUser = await Group.findOne({
+        where: { captainId: auth.userAccount.id },
+        attributes: ['id'],
+      })
+    }
+
+    sendProfile(response, auth.userAccount, auth.isAdmin, groupForUser?.id)
```

Comment:
I'd love some feedback on this because there's probably a better way to do this. The goal is to send down the user's `groupId` (if they have one) so that the UI can figure out whether they need onboarding or not.

Reply:
The incongruity here is that in the backend there's a one-to-many association between user accounts and groups. I don't know if we need/want to support a person being a member of multiple groups. If we do it'd likely complicate the UI in addition to code you've added in this PR.

The code as-written is mostly fine so long as we don't somehow end up with a user with multiple groups in the DB, which is possible at least as far as the backend is concerned.

I'd be ok with more or less leaving it as is while we gather feedback and pay the tech debt in the future if need be to fully support one-to-many rather than just one-to-one. Short term, if we *don't* need to support one-to-many we should add validation checks in `src/resolvers/group.ts` to ensure it doesn't happen since right now I think the admin could update multiple groups to have the same captain..

Reply:
I appreciate your thought process. I agree with shipping this as-is and figuring out the one-to-many vs one-to-one relationship in the future. I'll create an issue and add a comment in the code.